ansi2/src/lex.rs

```diff
@@ -7,72 +7,74 @@ use nom::sequence::tuple;
 
 use nom::IResult;
 
-use crate::theme::ColorTable;
+use crate::theme::{ColorTable, COLOR256};
 
 #[derive(Debug, Clone, Copy)]
 pub enum AnsiColor {
     Color8(u32),
+    Color256(u32),
     Rgb(u32, u32, u32),
 }
 impl AnsiColor {
     pub fn name(&self) -> String {
         match self {
-            AnsiColor::Color8(n) => {
-                match n {
-                    30 | 40 => "black".into(),
-                    31 | 41 => "red".into(),
-                    32 | 42 => "green".into(),
-                    33 | 43 => "yellow".into(),
-                    34 | 44 => "blue".into(),
-                    35 | 45 => "magenta".into(),
-                    36 | 46 => "cyan".into(),
-                    37 | 47 => "white".into(),
-
-                    90 | 100 => "bright_black".into(),
-                    91 | 101 => "bright_red".into(),
-                    92 | 102 => "bright_green".into(),
-                    93 | 103 => "bright_yellow".into(),
-                    94 | 104 => "bright_blue".into(),
-                    95 | 105 => "bright_magenta".into(),
-                    96 | 106 => "bright_cyan".into(),
-                    97 | 107 => "bright_white".into(),
-                    _ => "white".into(),
-                }
-            }
+            AnsiColor::Color8(n) => match n {
+                30 | 40 => "black".into(),
+                31 | 41 => "red".into(),
+                32 | 42 => "green".into(),
+                33 | 43 => "yellow".into(),
+                34 | 44 => "blue".into(),
+                35 | 45 => "magenta".into(),
+                36 | 46 => "cyan".into(),
+                37 | 47 => "white".into(),
```

Comment:
39 and 49 should be white by default. If you have a specific use case, feel free to open a new issue. This PR is already larger than I expected :D.

Reply:
The point of default is that your terminal knows what text color contrasts well with your background color. So, on light terminals the text is dark and on dark terminals the text is light. I'm not sure exactly how it works because it's not necessarily index 0 (black) or 15 (white) for the color. On Dark Solarized, which has a kind of green background, it chooses a gray color, which is index 10, 11, or 12. I can submit another issue on this subject. Thanks.